lib/modules/manager/npm/extract/index.ts

```diff
@@ -271,3 +293,19 @@ export async function extractAllPackageFiles(
   await postExtract(npmFiles);
   return npmFiles;
 }
+
+async function resolvePackageManager(packageFile: string): Promise<boolean> {
+  let hasPkgManager = false;
+  const packageJson = await findLocalSiblingOrParent(
+    packageFile,
+    'package.json',
+  );
+
+  if (packageJson) {
+    const content = await readLocalFile(packageJson, 'utf8');
+
+    hasPkgManager = content ? hasPackageManager(content) : false;
```

Comment:
see other comment. use existing zod schema

Reply:
use the zod schema now inside the hasPackageManager function

Reply:
reuse https://github.com/renovatebot/renovate/blob/b084fe8faa248a2890fa2cdf4ab32648f6e929f5/lib/modules/manager/npm/post-update/utils.ts#L17-L29